spec.html

```diff
@@ -24176,16 +24390,392 @@ <h1>Runtime Semantics: Evaluation</h1>
+          1. If _metadataRecord_.[[Public]] is not an empty List, then
+            1. Let _parentPublicMetadata_ be ? Get(_parentMetadataObjForKey_, *"public"*).
+            1. If _parentPublicMetadata_ is *undefined*, set _parentPublicMetadata_ to %Object.prototype%.
+            1. Let _publicMetadata_ be OrdinaryObjectCreate(_parentPublicMetadata_).
+            1. For each element _meta_ of _metadataRecord_.[[Public]], do
+              1. Let _key_ be _meta_.[[Key]].
+              1. Let _value_ be _meta_.[[Value]].
+              1. Perform ! CreateDataPropertyOrThrow(_publicMetadata_, _key_, _value_).
+            1. Perform ! CreateDataPropertyOrThrow(_metadataObjForKey_, *"public"*, _publicMetadata_).
+          1. If _metadataRecord_.[[Private]] is not an empty List, then
+            1. Let _parentPrivateMetadata_ be ? Get(_parentMetadataObjForKey_, *"private"*).
+            1. If _parentPrivateMetadata_ is *undefined*, let _privateMetadata_ be a new empty List.
+            1. Else, let _privateMetadata_ be ? CreateListFromArrayLike(_parentPrivateMetadata_).
+            1. For each element _meta_ in _metadataRecord_.[[Private]], do
+              1. Append _meta_.[[Value]] to _privateMetadata_.
+            1. Let _privateMetadataArray_ be CreateArrayFromList(_privateMetadata_).
+            1. Perform ! CreateDataPropertyOrThrow(_metadataObjForKey_, *"private"* , _privateMetadataArray_).
+          1. Perform ! CreateDataPropertyOrThrow(_metadataObj_, _key_, _metadataObjForKey_).
+        1. Perform ! CreateDataPropertyOrThrow(_targetObject_, @@metadata, _metadataObj_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratorcontextobject" type="abstract operation">
+      <h1>
+        CreateDecoratorContextObject (
+          _kind_: ~class~, ~field~, ~accessor~, ~method~, ~getter~, or ~setter~,
+          _name_: a String, Symbol, or Private Name,
+          _metadataList_: a List of Metadata Records,
+          _initializers_: a List of function objects,
+          _decorationState_: a Record with a field [[Finished]] (a Boolean),
+          optional _isStatic_: a Boolean,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns a context object which is passed as the second argument to decorators of the given class element.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _contextObj_ be OrdinaryObjectCreate(%Object.prototype%).
+        1. If _kind_ is ~method~, let _kindStr_ *"method"*.
+        1. Else if _kind_ is ~getter~, let _kindStr_ be *"getter"*.
+        1. Else if _kind_ is ~setter~, let _kindStr_ be *"setter"*.
+        1. Else if _kind_ is ~accessor~, let _kindStr_ be *"accessor"*.
+        1. Else if _kind_ is ~field~, let _kindStr_ be *"field"*.
+        1. Else,
+          1. Assert: _kind_ is ~class~.
+          1. Let _kindStr_ be *"class"*.
```

Comment:
even if this logic is only used in one place, i lightly think it might deserve its own AO?